prov/util/src/util_shm.c

```diff
@@ -62,6 +62,25 @@ static void smr_peer_addr_init(struct smr_addr *peer)
 	peer->addr = FI_ADDR_UNSPEC;
 }
 
+void smr_cma_check(struct smr_region *smr, struct smr_region *peer_region)
+{
+	struct iovec local_iov;
+	int peer_pid = 0;
+	int ret;
+
+	local_iov.iov_base = &peer_pid;
+	local_iov.iov_len = sizeof(peer_pid);
+	ret = process_vm_readv(peer_region->pid, &local_iov, 1,
+			       &peer_region->remote_iov, 1, 0);
+	if (ret == -1) {
+		smr->cma_cap = 0;
+	} else {
+		assert(peer_pid == peer_region->pid);
+		smr->cma_cap = 1;
+		peer_region->cma_cap = 1;
+	}
+}
+
```

Comment:
I don't see the need to store the remote_iov here or writing to the pid.
We should just be able to write directly to the peer_smr->cma_cap and use a simple address translation if we just save the smr base address.
ie:
Initialize
```
smr->cma_cap = -1;
smr->base_addr = *smr;
```
And then here do something like this
```
smr_cma_check(...)
{
    local_iov.iov_base = &smr->cma_cap;
    remote_iov.iov_base = peer_smr->base_addr + (peer_smr->cma_cap - peer_smr);
    process_vm_writev(peer_smr->pid, &local_iov, 1, &remove_iov, 1);
}
```

Reply:
Good point. Fixed in the new revision. Thank you.